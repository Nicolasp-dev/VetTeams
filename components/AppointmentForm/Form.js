import React from "react";
import ErrorFormMsg from "../ErrorFormMsg/ErrorFormMsg";
// 3rd Party Library
import { useForm } from "react-hook-form";
import { useContext } from "react";
import { GeneralContext } from "../../contexts/GeneralContext";
import { feedbackAlert } from "../../helpers/alerts/alerts";
// ---------- //

const Form = () => {
  const {
    handleSubmit,
    register,
    formState: { errors },
  } = useForm();

  const { appointmentCtx } = useContext(GeneralContext);
  const { addAppointmentHandler } = appointmentCtx;

  const onSubmitHandler = (values) => {
    addAppointmentHandler(values);
    feedbackAlert(values.service, "has been register successfully", "success");
  };

  return (
    <div className=" flex flex-col gap-y-6 sm:py-5 py-[110px] px-5 sm:pl-[7rem] lg:pl-0">
      <h1 className="text-2xl sm:text-[2rem] text-[#1F6B6B] font-bold text-center">
        Make an appointment
      </h1>

      <p className="text-sm sm:text-md text-justify">
        In this section you can request an appointment for a general medical
        consultation, specialized medical consultation, vaccinations and
        administrative procedures.
        <br />
        <br />
        It is not possible to request appointments online for some activities
        (diagnostic tests, minor surgery, etc.) that require a prior indication
        from a professional or special preparation conditions.
      </p>

      <form
        className="text-center flex flex-col"
        onSubmit={handleSubmit(onSubmitHandler)}
      >
        <fieldset className="grid grid-cols-1 lg:grid-cols-2 gap-4 mb-4">
          <div>
            <select
              className="placeholder:text-mid-green block bg-transparent w-full
                border-2 border-mid-green rounded-md py-2 pl-2 sm:pl-4 pr-3
                focus:outline-none focus:border-mid-green focus:ring-mid-green
                focus:ring-1 max-h-[45px]"
              type="text"
              placeholder="Service"
              {...register("service", {
                required: { value: true, message: "*Required field" },
              })}
            >
              <option>Medical Service</option>
              <option>Grooming Service</option>
            </select>
            {errors.service && (
              <ErrorFormMsg errorMessage={errors.service?.message} />
            )}
          </div>
          <div>
            <input
              className="placeholder:text-mid-green block bg-transparent w-full border-2 border-mid-green rounded-md py-2 pl-2  sm:pl-4 pr-3  focus:outline-none focus:border-mid-green focus:ring-mid-green focus:ring-1"
              type="date"
              placeholder="Date"
              {...register("date", {
                required: { value: true, message: "*Required field" },
              })}
            />
            {errors.date && (
              <ErrorFormMsg errorMessage={errors.date?.message} />
            )}
          </div>
        </fieldset>

        <textarea
          className="placeholder:text-mid-green block bg-transparent w-full border-2 border-mid-green rounded-md py-2 pl-4 focus:outline-none focus:border-mid-green focus:ring-mid-green focus:ring-1 mb-5"
          cols={10}
          rows={6}
          placeholder="Some description"
          {...register("description", { required: true })}
        ></textarea>

        <button
          type="submit"
          className="px-2 rounded-md h-[3rem]  bg-mid-green text-white tracking-wider text-sm sm:text-md"
        >
          Submit
        </button>
      </form>
    </div>
  );
};

export default Form;
