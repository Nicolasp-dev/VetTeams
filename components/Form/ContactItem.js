import React from "react";

const ContactItem = (props) => {
  return (
    <li className="flex gap-5 items-center mb-3 text-lg">{props.children}</li>
  );
};

export default ContactItem;
