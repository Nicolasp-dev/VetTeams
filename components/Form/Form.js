import React from "react";
// 3rd PArty Library
import { useForm } from "react-hook-form";
// Component
// import InputForm from "./InputForm";
// -------------------------------------------------- //

const Form = () => {
  //prettier-ignore
  const { handleSubmit, register, formState: { errors } } = useForm();

  const submitHandler = (values) => {
    console.log(values);
  };

  return (
    <div className="col-start-2 col-end-4 flex flex-col items-center justify-center">
      <h1 className="text-4xl text-[#1F6B6B] font-semibold text-center">
        Make an appointment
      </h1>

      <form
        className="w-[80%] text-center py-8 relative "
        onSubmit={handleSubmit(submitHandler)}
      >
        <fieldset className="grid grid-cols-2 gap-4 mb-4">
          <div>
            <input
              className="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-9 pr-3 focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1 max-h-[45px]"
              type="text"
              placeholder="Patient's name"
              {...register("patient", { required: true })}
            />
          </div>
          <div>
            <select
              className="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-9 pr-3  focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1 max-h-[45px]"
              type="text"
              placeholder="Species"
              {...register("species", { required: true })}
            >
              <option>Canine</option>
              <option>Feline</option>
            </select>
          </div>
          <div>
            <input
              className="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-9 pr-3  focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1 max-h-[45px]"
              type="number"
              placeholder={errors.age ? "*Age field is required" : "Age"}
              {...register("age", { required: true })}
            />
          </div>
          <div>
            <select
              className="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-9 pr-3  focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1 max-h-[45px]"
              type="text"
              placeholder="Patient gender"
              {...register("gender", { required: true })}
            >
              <option className="text-[#1F6B6B]">Male</option>
              <option>Female</option>
            </select>
          </div>
          <div>
            <input
              className="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-9 pr-3  focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1 max-h-[45px]"
              type="text"
              placeholder="Tutor's name"
              {...register("tutor", { required: true })}
            />
          </div>
          <div>
            <input
              className="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-9 pr-3  focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1 max-h-[45px]"
              type="text"
              placeholder="Specialist"
              {...register("specialist", { required: true })}
            />
          </div>
          <div>
            <input
              className="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-9 pr-3  focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1 max-h-[45px]"
              type="date"
              placeholder="Date"
              {...register("date", { required: true })}
            />
          </div>
          <div>
            <input
              className="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-9 pr-3  focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1 max-h-[45px]"
              type="number"
              placeholder="Phone"
              {...register("phone", { required: true })}
            />
          </div>
        </fieldset>

        <textarea
          className="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-9 pr-3  focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1 mb-5"
          cols={10}
          rows={6}
          placeholder="Some description"
          {...register("text-area", { required: true })}
        ></textarea>

        {/* prettier-ignore */}
        <button type="submit" className="px-2 rounded-3xl h-[3rem] w-[8rem] bg-[#1F6B6B] text-white right-3 absolute tracking-wider">Submit</button>
      </form>
    </div>
  );
};

export default Form;
