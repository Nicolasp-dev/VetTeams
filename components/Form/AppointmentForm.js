import React from "react";
import Image from "next/image";
// Assets
import Yorky from "../../public/assets/Yorkshire-Terrier-PNG-Image.png";
import { BsTelephone } from "react-icons/bs";
import { MdOutlineEmergency } from "react-icons/md";
import { Ri24HoursFill } from "react-icons/ri";
import { TiLocation } from "react-icons/ti";
// Components
import ContactItem from "./ContactItem";
import Form from "./Form";

const AppointmentForm = () => {
  // const submitErrorHandler = (error) => {
  //   console.log(error);
  // };

  return (
    <section className="w-full grid grid-cols-3 overflow-hidden">
      {/* QUICK CONTACT  */}
      <div className="h-screen bg-[#1F6B6B] col-start-1 col-end-2 flex flex-col justify-evenly items-center py-10">
        <div className="w-[80%] flex flex-col items-center">
          <h1 className="text-white text-3xl font-semibold mb-5">
            Quick Contact
          </h1>
          <ul className="text-white">
            <ContactItem>
              <BsTelephone size={20} />
              <p>+57 (000) 000-000</p>
            </ContactItem>
            <ContactItem>
              <MdOutlineEmergency size={22} />
              <p>Contact Vet-Teams</p>
            </ContactItem>
            <ContactItem>
              <Ri24HoursFill size={22} />
              <p>Open 24/7</p>
            </ContactItem>
            <ContactItem>
              <TiLocation size={25} />
              <p>Some address - 123</p>
            </ContactItem>
          </ul>
        </div>
        <div>
          <Image src={Yorky} />
        </div>
      </div>

      {/* FORM */}
      <Form />
    </section>
  );
};

export default AppointmentForm;
