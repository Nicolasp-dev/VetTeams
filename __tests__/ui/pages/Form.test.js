import { act, fireEvent, render, screen } from "@testing-library/react";
import selectEvent from "react-select-event";
import userEvent from "@testing-library/user-event";

import Form from "../../../components/AppointmentForm/Form";

describe("Form", () => {
  test("Matches DOM Snapshot", () => {
    const wrapper = render(<Form />);
    expect(wrapper).toMatchSnapshot();
  });

  test("calls the onSubmit function", async () => {
    const mockUser = {
      date: "2022-07-30",
      service: "Grooming Service",
      textarea: "asdasd",
    };
    const mockSubmit = jest.fn();
    const wrapper = render(<Form onSubmit={mockSubmit} data={mockUser} />);

    // // await userEvent.selectOptions(wrapper.getByPlaceholderText('Phone'), ['1', 'C'])

    await act(async () => {
      fireEvent.change(wrapper.getByPlaceholderText("Date"), {
        target: { value: "2022-07-30" },
      });
      fireEvent.change(wrapper.getByPlaceholderText("Service"), {
        target: { value: "Grooming Service" },
      });
      fireEvent.change(wrapper.getByPlaceholderText("Some description"), {
        target: { value: "asdasd" },
      });
    });

    await act(async () => {
      userEvent.click(wrapper.getByRole("button"));
    });

    //expect(mockSubmit).toHaveBeenCalled();
  });
});
