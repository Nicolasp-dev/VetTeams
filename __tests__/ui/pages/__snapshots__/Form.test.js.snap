// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Matches DOM Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full col-span-2 mb-[5rem] mt-[100px] md:col-start-1 md:col-end-3 flex justify-center items-center row-start-1 row-end-3"
      >
        <div
          class="w-[70%] flex flex-col gap-y-6 "
        >
          <h1
            class="text-4xl text-[#1F6B6B] font-semibold text-center"
          >
            Make an appointment
          </h1>
          <p
            class="text-justify"
          >
            In this section you can request an appointment for a general medical consultation, specialized medical consultation, vaccinations and administrative procedures.
            <br />
            <br />
            It is not possible to request appointments online for some activities (diagnostic tests, minor surgery, etc.) that require a prior indication from a professional or special preparation conditions.
          </p>
          <form
            class="text-center relative "
            data-testid="form-element"
          >
            <fieldset
              class="grid grid-cols-1 lg:grid-cols-2 gap-4 mb-4"
            >
              <div>
                <select
                  class="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-2 sm:pl-4 pr-3 focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1 max-h-[45px]"
                  name="service"
                  placeholder="Service"
                  type="number"
                >
                  <option>
                    Medical Service
                  </option>
                  <option>
                    Grooming Service
                  </option>
                </select>
              </div>
              <div>
                <input
                  class="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-2 sm:pl-4 pr-3 focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1"
                  name="date"
                  placeholder="Date"
                  type="date"
                />
              </div>
            </fieldset>
            <textarea
              class="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-9 pr-3 focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1 mb-5"
              cols="10"
              name="textarea"
              placeholder="Some description"
              rows="6"
            />
            <button
              class="px-2 rounded-3xl h-[3rem] w-[8rem] bg-[#1F6B6B] text-white right-3 absolute tracking-wider"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full col-span-2 mb-[5rem] mt-[100px] md:col-start-1 md:col-end-3 flex justify-center items-center row-start-1 row-end-3"
    >
      <div
        class="w-[70%] flex flex-col gap-y-6 "
      >
        <h1
          class="text-4xl text-[#1F6B6B] font-semibold text-center"
        >
          Make an appointment
        </h1>
        <p
          class="text-justify"
        >
          In this section you can request an appointment for a general medical consultation, specialized medical consultation, vaccinations and administrative procedures.
          <br />
          <br />
          It is not possible to request appointments online for some activities (diagnostic tests, minor surgery, etc.) that require a prior indication from a professional or special preparation conditions.
        </p>
        <form
          class="text-center relative "
          data-testid="form-element"
        >
          <fieldset
            class="grid grid-cols-1 lg:grid-cols-2 gap-4 mb-4"
          >
            <div>
              <select
                class="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-2 sm:pl-4 pr-3 focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1 max-h-[45px]"
                name="service"
                placeholder="Service"
                type="number"
              >
                <option>
                  Medical Service
                </option>
                <option>
                  Grooming Service
                </option>
              </select>
            </div>
            <div>
              <input
                class="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-2 sm:pl-4 pr-3 focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1"
                name="date"
                placeholder="Date"
                type="date"
              />
            </div>
          </fieldset>
          <textarea
            class="placeholder:text-[#1F6B6B] block bg-white w-full border-2 border-[#1F6B6B] rounded-2xl py-2 pl-9 pr-3 focus:outline-none focus:border-[#1F6B6B] focus:ring-[#1F6B6B] focus:ring-1 mb-5"
            cols="10"
            name="textarea"
            placeholder="Some description"
            rows="6"
          />
          <button
            class="px-2 rounded-3xl h-[3rem] w-[8rem] bg-[#1F6B6B] text-white right-3 absolute tracking-wider"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
